From 569df9cc8855bfe79cee06fd29d6e49d02ab27a0 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Mon, 27 Feb 2017 16:25:00 +0800
Subject: [PATCH 1685/5242] MLK-14275: regulator: pf1550-regulator-rpmsg:
 clear 'u32 val' before read value by rpmsg

commit  dcee753abe5f858dfdd1d30eab37318d14621bc1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Actually, m4 only fill the least 1byte, so we'd better clear 'val' before
reading by rpmsg, thus the gabage data will not bother us.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
(cherry picked from commit df3428c6588a00fa153a3b6996cc33b21b69efb6)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/pf1550-regulator-rpmsg.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/regulator/pf1550-regulator-rpmsg.c b/drivers/regulator/pf1550-regulator-rpmsg.c
index 57404ea..11a8933 100644
--- a/drivers/regulator/pf1550-regulator-rpmsg.c
+++ b/drivers/regulator/pf1550-regulator-rpmsg.c
@@ -382,6 +382,7 @@ static ssize_t pf1550_registers_show(struct device *dev,
 
 		msg.header.cmd = PF1550_GET_REG;
 		msg.reg = i;
+		msg.val = 0;
 
 		err = pf1550_send_message(&msg, info);
 		if (err)
-- 
1.7.9.5

